common/djangoapps/course_modes/models.py

```diff
@@ -155,6 +155,21 @@ def modes_for_course(cls, course_id):
         return modes
 
     @classmethod
+    def create_course_mode(cls, course_id, min_price, mode_slug, currency, mode_display_name=DEFAULT_MODE_SLUG):
```

Comment:
Perhaps not necessary, but we do have a set of known modes, might be nice to validate we're creating a known mode.